### PR TITLE
Add VDT IT support:  updated vulnerable package not reported as solved

### PR DESCRIPTION
**Pull request by Deblintrake09**
|Related issue|
|-------------|
| #4045            |

## Description

This Issue aims to add IT support to verify when a package is vulnerable to a CVE and updated to a version that is still vulnerable, there is no alert showing the vulnerability as solved

### Added

- Test module `test_scan_updated_package_still_vulnerable.py`

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @Deblintrake09 (Developer)  | test_vulnerability_detector           | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/41973/)[:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/41974/)[:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/41972/)  | ⚫⚫⚫ | Centos manager         |  https://github.com/wazuh/wazuh-qa/pull/4389/commits/93d5702edf03c60bc5c3158c4fd901aabd0fe043       | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### CODE REVIEW

File: tests/integration/test_vulnerability_detector/data/feeds/canonical/custom_feed_updated_packace_still_vulnerable.xml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        <linux-def:dpkginfo_object id="oval:com.ubuntu.focal:obj:1" version="1" comment="The 'custom-package-0' package binary.">
+            <linux-def:name>custom-package-0</linux-def:name>
+        </linux-def:dpkginfo_object>
+    </objects>
+    <states>
+        <linux-def:dpkginfo_state id="oval:com.ubuntu.focal:ste:1" version="1" comment="The package version is less than '10.0.0'.">
+            <linux-def:evr datatype="debian_evr_string" operation="less than">10.0.0</linux-def:evr>
+        </linux-def:dpkginfo_state>
+    </states>
+</oval_definitions>
```
Comment (mauromalara) [inline tests/integration/test_vulnerability_detector/data/feeds/canonical/custom_feed_updated_packace_still_vulnerable.xml:51]: Add endline

File: tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_scan_updated_package_still_vulnerable.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+- name: no_alert_updated_package_still_vulnerable
```
Comment (mauromalara) [inline tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_scan_updated_package_still_vulnerable.yaml:1]: Change the name to a shorter one with capital letters. I recommend: `VULN_REMAINS_IN_PACKAGE` or `PACKAGE_STILL_VULNERABLE`

File: tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_scan_updated_package_still_vulnerable.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  description: When a package is updated to still vulnerable version, no alert appears showing package vuln as solved
```
Comment (mauromalara) [inline tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_scan_updated_package_still_vulnerable.yaml:2]: I think the following description will be more accurate:

```
Verify that when updating a vulnerable package to another version that is still vulnerable to the same CVE, the system triggers an alert for the new package installed but does not generate an alert notifying that the vulnerability was solved.
```

File: tests/integration/test_vulnerability_detector/data/feeds/canonical/custom_feed_updated_packace_still_vulnerable.xml (hunk 1 above)
Comment (BelenValdivia) [inline tests/integration/test_vulnerability_detector/data/feeds/canonical/custom_feed_updated_packace_still_vulnerable.xml:51]: Add end line

**Review by BelenValdivia — CHANGES_REQUESTED** [review]

**Review by mauromalara — CHANGES_REQUESTED** [review]
Great Job! 👏🏻 

🗒️ Only a few changes are required.
⚠️ Please, run the tests in Jenkins with the updated branch after applying said changes.

File: tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_scan_updated_package_still_vulnerable.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  configuration_parameters: null
+  metadata:
+    provider_name: canonical
+    system: FOCAL
+    oval_feed: custom_feed_updated_packace_still_vulnerable.xml
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_scan_updated_package_still_vulnerable.yaml]: Done in https://github.com/wazuh/wazuh-qa/pull/4389/commits/308bc8c4b492b99c117039f889eef100fbdd668d

File: tests/integration/test_vulnerability_detector/data/feeds/canonical/custom_feed_updated_packace_still_vulnerable.xml (hunk 1 above)
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/data/feeds/canonical/custom_feed_updated_packace_still_vulnerable.xml]: Done in https://github.com/wazuh/wazuh-qa/pull/4389/commits/308bc8c4b492b99c117039f889eef100fbdd668d

File: tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_scan_updated_package_still_vulnerable.yaml (hunk 2 above)
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_scan_updated_package_still_vulnerable.yaml]: Done in https://github.com/wazuh/wazuh-qa/pull/4389/commits/18fa7b86f2c5959cfc8fd859de82e4f5ad17bd2a

File: tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_scan_updated_package_still_vulnerable.yaml (hunk 3 above)
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_scan_updated_package_still_vulnerable.yaml]: Done in https://github.com/wazuh/wazuh-qa/pull/4389/commits/18fa7b86f2c5959cfc8fd859de82e4f5ad17bd2a

**Review by mauromalara — APPROVED** [review]
LGTM! 🚀

**Comment by BelenValdivia** [comment]
LGTM!

**Review by BelenValdivia — APPROVED** [review]

**Comment by Deblintrake09** [comment]
This development has already been merged into Master, new target. So no new PR needed.

Merged in https://github.com/wazuh/wazuh-qa/commit/ea84bee457e4328e3115604b263f6dad7c154cee

**Pull request merged**